### PR TITLE
Feature server complete habit

### PR DESCRIPTION
**Pull request by TakenouchiTR**
Allows the user to complete habits and earn coins, including bonuses for completing all habits of a given frequency.

### CODE REVIEW

**Review by Steven-Carriger — CHANGES_REQUESTED** [review]
In the server/backend/habit.py the message in the Exception being raised when checking the frequency contains a small typo. Between is mispelled as beteen.  Other than that, everything looks wonderful. Excellent work!

File: server/backend/habit.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def frequency(self, frequency: int):
         if not isinstance(frequency, int):
             raise Exception("frequency must be an int")
         if frequency < CompletionFrequency.DAILY.value or frequency > CompletionFrequency.MONTHLY.value:
-            raise Exception("frequency must be betwwn 0 and 2, inclusive")
+            raise Exception("frequency must be beteen 0 and 2, inclusive")
```
Comment (Steven-Carriger) [inline server/backend/habit.py:163]: This is what I am referring to.

Comment (bfreder1) [reply server/backend/habit.py]: Spelling mistake got corrected, gonna close this convo.

**Review by bfreder1 — APPROVED** [review]
With the inclusion of the fixes to the tests and the spelling error, we're good to go with merging here. Glad to see the server working.

**Pull request merged**